network/builder/src/builder.rs

```diff
@@ -291,24 +287,24 @@ impl NetworkBuilder {
                 network_simple_onchain_discovery::gen_simple_discovery_reconfig_subscription();
             self.reconfig_subscriptions
                 .push(simple_discovery_reconfig_subscription);
-            self.configuration_change_listener_cfg = Some(ConfigurationChangeListenerConfig::new(
-                self.network_context.role(),
-                self.conn_mgr_reqs_tx().expect("We just set this value"),
-                simple_discovery_reconfig_rx,
-            ));
+            self.configuration_change_listener_builder =
+                Some(ConfigurationChangeListenerBuilder::create(
+                    self.network_context.role(),
+                    self.conn_mgr_reqs_tx().expect("We just set this value"),
+                    simple_discovery_reconfig_rx,
+                ));
         }
-
         self.build_connectivity_manager()
     }
 
     /// Build the ConnectivityManager from the provided configuration.
     fn build_connectivity_manager(&mut self) -> &mut Self {
         if let Some(ref mut connectivity_builder) = self.connectivity_manager_builder {
             connectivity_builder.build(&self.executor);
-            if let Some(config) = self.configuration_change_listener_cfg.take() {
-                let configuration_change_listener =
-                    network_simple_onchain_discovery::build_configuration_change_listener(config);
-                self.configuration_change_listener = Some(configuration_change_listener);
+            if let Some(ref mut change_listener_builder) =
+                self.configuration_change_listener_builder
+            {
+                change_listener_builder.build();
             }
```

Comment:
have some fun with rust:
`self.configuration_change_listener_builder.map(|c| c.build());`

Reply:
> have some fun with rust:
> `self.configuration_change_listener_builder.map(|c| c.build());`

that makes my head hurt.